app/controllers/api/public/v1/courses_controller.rb

```diff
@@ -7,24 +7,40 @@ class CoursesController < API::Public::V1::ApplicationController
         def index
           render jsonapi: paginate(courses),
                  include: include_param,
-                 meta: { count: courses.count('course.id') },
+                 meta: { count: cached_course_count },
                  class: API::Public::V1::SerializerService.call
         rescue ActiveRecord::StatementInvalid
           render json: { status: 400, message: 'Invalid changed_since value, the format should be an ISO8601 UTC timestamp, for example: `2019-01-01T12:01:00Z`' }.to_json, status: :bad_request
         end
 
         private
 
+        def cached_course_count
+          year = permitted_params[:recruitment_cycle_year] || RecruitmentCycle.current.year
+
+          if permitted_params[:no_cache]
```

Comment:
Below might be something for another PR.

It might be more conventional to read the `Cache-Control` request header and return cached vs non-cached results.

I haven't quite dug into if that's indeed the right header, or what the header value should be (I _think_ `no-cache`?), and also if my assumption is considered standard practice, but it might be worth looking into.

Reply:
I'm happy to remove this, I don't think it should be merged. Undocumented api params probably aren't a good idea.

As for the cache headers, I believe those (response headers) are for instructing how the client should cache, not how the server should respond.

Reply:
I _think_ that [this section](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#cache_directives) is indicating that there are request header standards too, but yeah, I haven't seen it in practice yet. I wonder how the browser implements the "Disable cache" checkbox.

But yeah, I agree. Let's remove the `no_cache` for now.